sunpy/map/mapbase.py

```diff
@@ -2460,9 +2487,30 @@ def plot(self, annotate=True, axes=None, title=True,
                 norm.vmax = imshow_args.pop('vmax')
 
         if self.mask is None:
-            ret = axes.imshow(self.data, **imshow_args)
+            data = self.data
+        else:
+            data = np.ma.array(np.asarray(self.data), mask=self.mask)
+
+        if autoalign == 'pcolormesh':
+            axes.set_aspect(getattr(imshow_args, 'aspect', 1))
+
+            # Remove imshow keyword arguments that are not accepted by pcolormesh
+            for item in ['aspect', 'extent', 'interpolation', 'origin']:
+                if item in imshow_args:
+                    del imshow_args[item]
```

Comment:
Similar to above, do we want to warn if these keyword arguments are removed?

Reply:
I am not really in favour of throwing warnings here.

Reply:
How about raising errors? From memory that's the approach `WCSAxes` takes in some similar situations. I'm not too fussed, but either way this should definitely be mentioned in the docstring.

Reply:
All of these arguments do not exist for `pcolormesh()` compared to `imshow()`.
* I handle `aspect` separately earlier, so removing it here is fine.
* `origin` and `extent` don't matter.  We don't reasonably expect any user to set any of these, but we set them internally.  (Ideally, the code should be reversed: they shouldn't be in `plot_settings` and instead be _added_ by `plot()` if `imshow()` is going to be used.)
* `interpolation` is an exception, because a user might ask for that, but `pcolormesh()` doesn't support it.  We can raise a warning if a user specifically asks for an interpolation method other than `'none'` or `'nearest'` so they know that they won't be getting what they ask for.

Reply:
👍

Reply:
Warning added for `interpolation`